lib/spack/spack/build_systems/cmake.py

```diff
@@ -34,56 +37,142 @@ def _extract_primary_generator(generator):
     return primary_generator
 
 
-class CMakePackage(PackageBase):
+class CMakePackage(spack.package_base.PackageBase):
     """Specialized class for packages built using CMake
 
     For more information on the CMake build system, see:
     https://cmake.org/cmake/help/latest/
+    """
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = "CMakePackage"
+
+    #: Legacy buildsystem attribute used to deserialize and install old specs
+    legacy_buildsystem = "cmake"
+
+    buildsystem("cmake")
+
+    with when("buildsystem=cmake"):
+        # https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
+        variant(
+            "build_type",
+            default="RelWithDebInfo",
+            description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
+        )
+        # CMAKE_INTERPROCEDURAL_OPTIMIZATION only exists for CMake >= 3.9
+        # https://cmake.org/cmake/help/latest/variable/CMAKE_INTERPROCEDURAL_OPTIMIZATION.html
+        variant(
+            "ipo",
+            default=False,
+            when="^cmake@3.9:",
+            description="CMake interprocedural optimization",
+        )
+        depends_on("cmake", type="build")
+        depends_on("ninja", type="build", when="platform=windows")
+
+    # FIXME (MULTIPLE BUILD SYSTEMS): Move to the corresponding builder?
+    def flags_to_build_system_args(self, flags):
```

Comment:
This flag handler probably should move to the build system. Are there issues with that?

I think all the flag handlers can be moved and if it comes to multiple build systems which want the same compiler wrapper insertions, then that can be handled by making a common superclass builder.

Reply:
I thought we decided to not move flag handlers for the time being? In any case, I'll have a look what it takes to move this method.

Reply:
For the record, decided in DM over slack to move this to a following PR